### PR TITLE
Issue #55: Initializing tm struct before use with mktime().

### PR DESCRIPTION
**Pull request by whiten01se**
Timestamp parsing behavior now consistent. Needs a test on a Raspberry.

### CODE REVIEW

File: src/util/time.cc [hunk 1]
```diff
@@ -22,6 +22,10 @@ std::string stringTime() {
 bool parseTime(const std::string& in, struct timespec* ts) {
   const char* buf = in.c_str();
   struct tm tm;
+
+  //initialize tm struct to 0s before using it, otherwise mktime will behave unpredictably - Kons
```
Comment (pietern) [inline src/util/time.cc]: Please use the same commenting style as in the rest of the code, with capitalization and a space after `//`.

File: src/util/time.cc [hunk 2]
```diff
@@ -42,6 +46,10 @@ bool parseTime(const std::string& in, struct timespec* ts) {
   }
 
   auto t = mktime(&tm);
+
+  //the resulting t should always be greater 0, otherwise there was a problem with date conversion
```
Comment (pietern) [inline src/util/time.cc]: Same as above.

Comment (whiten01se) [reply src/util/time.cc]: Done. :)

**Comment by pietern** [comment]
Thank you!

**Pull request merged**